engine/mesh.h

```diff
@@ -17,11 +18,14 @@ class Mesh
 public:
     virtual void Draw(const glm::vec3& position) const;
 
+    void SetColor(const GLRGBAColor& color);
+
 protected:
     VertexArray mVA {};
     VertexBuffer mVB {};
     IndexBuffer mIB {};
     Texture mTexture {};
+    GLRGBAColor mColor { 1.0f, 1.0f, 1.0f, 1.0f };
```

Comment:
I would possibly encourage a constant for white or similar, seems like something that will get reuse